### PR TITLE
feat: apply biome suggestions and fix few hooks

### LINKED ISSUE #2750: playlistMenu react component broken in 1.2.28
### 🔍 Have you checked Spicetify.app page for your issue?

- [X] I have checked the FAQ

### 🔍 Is there already an issue for your problem?

- [X] I have checked older issues, open and closed

### ℹ Environment / Computer Info

```Markdown
- Spotify version: 1.2.28
- Spicetify version: 2.29.1
```

### 📝 Description

playlist menu react component is broken in latest ver

### 📸 Screenshots

image

### PR DESCRIPTION
**Pull request by rxri**
Resolves #2750
This PR also fixes `DragHandler` availability from `1.2.1` to `1.2.8` and `extractColorPreset` on `1.2.5`

biome*: https://biomejs.dev/linter/rules

### CODE REVIEW

**Comment by SunsetTechuila** [comment]
Using `for of` instead of `foreach` in this case changes only one thing - performance (makes it worse)

**Comment by rxri** [comment]
> Using `for of` instead of `foreach` in this case changes only one thing - performance (makes it worse)

bro you either don't sit in js world or idk, for of is generally faster than foreach

File: jsHelper/spicetifyWrapper.js [hunk 1]
```diff
@@ -245,81 +245,76 @@ window.Spicetify = {
...
 			}
-		});
+		}
 		console.log(`${count}/${REACT_COMPONENT.length} Spicetify.ReactComponent methods and objects are OK.`);
 
 		count = REACT_CARD_COMPONENTS.length;
-		REACT_CARD_COMPONENTS.forEach(object => {
-			if (Spicetify.ReactComponent.Cards[object] === undefined || Spicetify.ReactComponent.Cards[object] === null) {
+		for (const method of REACT_CARD_COMPONENTS) {
+			if (Spicetify.ReactComponent.Cards[method] === undefined || Spicetify.ReactComponent.Cards[method] === null) {
```
Comment (SunsetTechuila) [inline jsHelper/spicetifyWrapper.js:276]: > react element
> method
hmm?

File: jsHelper/spicetifyWrapper.js [hunk 2]
```diff
@@ -245,81 +245,76 @@ window.Spicetify = {
...
 				console.error(`Spicetify.Player.${method} is not available. Please open an issue in the Spicetify repository to inform us about it.`);
 				count--;
 			}
-		});
+		}
 		console.log(`${count}/${PLAYER_METHOD.length} Spicetify.Player methods and objects are OK.`);
 
 		count = REACT_COMPONENT.length;
-		REACT_COMPONENT.forEach(method => {
+		for (const method of REACT_COMPONENT) {
```
Comment (SunsetTechuila) [inline jsHelper/spicetifyWrapper.js:266]: same as https://github.com/spicetify/spicetify-cli/pull/2751/files#r1443280409

File: jsHelper/spicetifyWrapper.js (hunk 1 above)
Comment (rxri) [reply jsHelper/spicetifyWrapper.js:276]: that's literally the same omg

**Review by SunsetTechuila — APPROVED** [review]

**Review by afonsojramos — APPROVED** [review]
Approved, but if we're changing to biome, which I support, we could probably fully move to biome, which means workflows and config.

**Comment by rxri** [comment]
Will do it in another pull request since it will take a while

**Pull request merged**